doc/sonic-application-extention/sonic-versioning-strategy.md

```diff
@@ -0,0 +1,117 @@
+- [Base OS API that a package uses](#base-os-api-that-a-package-uses)
+- [Open Questions](#open-questions)
+
+<!-- omit in toc -->
+## List of Figures
+
+### Revision
+
+| Rev |     Date    |       Author            | Change Description                   |
+|:---:|:-----------:|:-----------------------:|--------------------------------------|
+| 0.1 | 02/2021     | Stepan Blyshchak        | Initial Proposal                     |
+
+### Overview
+
+This document provides guidelines on how to do versioning for SONiC Docker Images using to semantic versioning strategy.
+
+<!-- omit in toc -->
+#### Motivation
+
+With new SONiC Application Extension Infrastructure SONiC Dockers and SONiC Base OS can be seperated and distributed individually.
+SONiC Dockers (aka SONiC Packages) can be installed, upgraded invididually from other. This creates a new problem which needs to be
+solved - compatibility between dependend Dockers and host OS. SONiC Application Extension Infrastructure provides a way to specify
+the package version dependency using semantic versioning (https://semver.org). This document provides a guideline on how to increment
+version numbers correctly on releases.  
+
+## SONiC Package API
+
+First of all, a clear definition of what is package API needs to be provided:
+
+SONiC package API is Redis DB interface including:
+  - CONFIG DB, APPL DB, STATE DB tables schema provided by this package
+  - Redis-based IPC Communication API: libswsscommon, libsairedis
+
+If any other kind of API is exposed by the SONiC Package it should be accounted as package API.
+
+## SONiC Package Releases
+
+As of today (202012 release), all SONiC Docker Images are released together with SONiC OS. With SONiC Application Extension
+Infrastructure SONiC Dockers are released independently. On every release an increment in package version number is required.
+The exact number in version that needs to be incremented depends on the type of changes comparing to previously released
+package.
+
+Types of changes:
+  - Breaking change: backwards incompatible change; reflects in ***major*** version number
+  - New functionality: backward compatible change; reflects in ***minor*** version number
+  - Bug fixes or enhancement; reflects in ***patch*** version number
+
+## Conventional Commits
+
+In order to help ***package maintainers*** to understand whether a breaking change was introduced
+comparing to previous release or a new functionality was included all SONiC repositories *can*
+follow conventional commits ():
+
+e.g:
+
+```
+feat: Introduce new methods in ConsumerTable
+
+BREAKING CHANGE: this feature breaks the Consumer/Producer based IPC
+```
+
+## SONiC Packages Versioning
+
+- Package release ***happens on SONiC release/branch-out*** or on demand for single Docker image: ***sub-release***
+- Manual version update ***is required*** when SONiC Package releases
+- Within a release major and minor version ***must*** not change
```

Comment:
Just to aid clarity - Within a SONIC release, the major and minor versions of the feature package ***must*** not change, i.e., any newer version of the feature package for that SONIC release, remains fully backward compatible.

Reply:
Yes, only bug fixes are allowed as usually for release branches